group/assignment6/RearrrangingCars.java

```diff
@@ -1,71 +1,85 @@
-	 * repeat that until reaching emptySlots
-	 * finally check if the start state equal the end,
-	 * if not it will be the case that at the beginning the emptySlot at the same position in two states
-	 * in this case generate in the first move a random slot to swap it with the empty 
-	 * and then starting the algorithm
-	 * @return linkedList of sequences to define moves as car, starting and ending spaces
+	 * given an starting and ending state of parking start with the parking
+	 * emptySlot in start state, get the car that will exist at the same slot in
+	 * the end state and do a move by swap the car from its current slot to the
+	 * emptySlot repeat that until reaching emptySlots and meanwhile check if the
+	 * start state equal the end before ending the moves, if not it will be the case that at the
+	 * beginning the emptySlot at the same position in two states in this case
+	 * generate in the first move a random slot to swap it with the empty and
+	 * then starting the algorithm
+	 * 
+	 * @return linkedList of sequences to define moves as car, starting and
+	 *         ending spaces
 	 */
 	public LinkedList<Sequence> generateMoves() {
 		LinkedList<Sequence> moves = new LinkedList<>();
 		char emptySlot = start.get(noCarKey);
 		move(moves, noCarKey, emptySlot);
-		if (!start.equals(end)) {
-			int key = rand.nextInt(start.keySet().size() - 1) + 1;
-			char value = start.get(key);
-			moves.add(new Sequence(key, value, emptySlot));
-			start.replace(noCarKey, start.get(key));
-			start.replace(key, emptySlot);
-			move(moves, key, value);
-		}
 		return moves;
 	}
 
 	private void move(LinkedList<Sequence> moves, int emptyKey, char emptySlot) {
-		int key = findKey(end, emptySlot);
-		char value = start.get(key);
-		if (emptySlot == value) {
+		int carID = findCarID(end, emptySlot);
+		if (carID == 0 && !start.equals(end)) {
+			carID = generateRandomCarID();
+		} else {
+			carsInRightSlots.add(carID);
+		}
+		char parkingSlot = start.get(carID);
+		if (emptySlot == parkingSlot) {
 			return;
 		}
-		moves.add(new Sequence(key, value, emptySlot));
-		start.replace(noCarKey, start.get(key));
-		start.replace(key, emptySlot);
-		move(moves, key, value);
+		moves.add(new Sequence(carID, parkingSlot, emptySlot));
+		start.replace(noCarKey, start.get(carID));
+		start.replace(carID, emptySlot);
+		move(moves, carID, parkingSlot);
 	}
 
-	private int findKey(Hashtable<Integer, Character> end, Character value) {
-		for (Entry<Integer, Character> element : end.entrySet()) {
-			if (element.getValue().equals(value)) {
+	private int findCarID(Hashtable<Integer, Character> configuration,
+			Character parkingSlot) {
+		for (Entry<Integer, Character> element : configuration.entrySet()) {
+			if (element.getValue().equals(parkingSlot)) {
 				return element.getKey();
 			}
 		}
-		return noExistKey;
+		throw new Error("That input is not correct, Missing car or slot");
+	}
+
+	private int generateRandomCarID() {

```

Comment:
There isn't any randomness involved here anymore, so maybe change the name or actually generate a random number rather than iterating through the array.